drivers/usb/udc/udc_it82xx2.c

```diff
@@ -1119,6 +1118,23 @@ static inline int work_handler_out(const struct device *dev, uint8_t ep)
 
 	it82xx2_xfer_out_data(dev, ep, buf);
 
+	LOG_DBG("Handle data OUT, %zu | %zu", len, net_buf_tailroom(buf));
+
+	if (net_buf_tailroom(buf) && len == udc_mps_ep_size(ep_cfg)) {
+		work_handler_xfer_continue(dev, ep, buf);
+		if (ep != USB_CONTROL_EP_OUT) {
+			err = udc_submit_ep_event(dev, buf, 0);
+		}
+		return err;
+	}
```

Comment:
That looks suspicious. Does it start a new transaction from the thread context?

Reply:
Yes, is there any risk in starting a new OUT transaction within thread? Perhaps there's something I'm not fully understanding.

Reply:
There is actually no need, it slows the transfer. The usual approach is to start the first transaction from the thread context, handle any subsequent transactions in the ISR context, and notify the thread when finished.

Reply:
Thanks for the feedback. This is now on our TODO list, and we will submit a PR if we make any improvements.